### PR TITLE
feat(exex, node): check ExEx head against the node head

### PR DESCRIPTION
**Pull request by shekhirin**
Ref https://github.com/paradigmxyz/reth/issues/10489

### CODE REVIEW

File: crates/node/builder/src/builder/mod.rs [hunk 1]
```diff
@@ -437,7 +437,7 @@ where
     pub fn install_exex<F, R, E>(self, exex_id: impl Into<String>, exex: F) -> Self
     where
         F: FnOnce(ExExContext<NodeAdapter<T, CB::Components>>) -> R + Send + 'static,
-        R: Future<Output = eyre::Result<E>> + Send,
+        R: Future<Output = eyre::Result<(Option<BlockHash>, E)>> + Send,
```
Comment (mattsse) [inline crates/node/builder/src/builder/mod.rs:440]: why does this take just the block hash?

File: crates/exex/exex/src/lib.rs [hunk 2]
```diff
@@ -49,3 +49,18 @@ pub use manager::*;
...
 pub use reth_exex_types::*;
+
+use reth_chainspec::Head;
+use reth_primitives::{BlockHash, BlockNumber};
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct ExExHead {
+    pub number: BlockNumber,
+    pub hash: BlockHash,
+}
```
Comment (mattsse) [inline crates/exex/exex/src/lib.rs:60]: this needs proper docs,

do we think we will eventually require additional info from the exex?
if so we could perhaps change this to

```rust
struct ExExInfo {
   head: Option<ExExHead>
}
```

**Review by mattsse — COMMENTED** [review]
I think they're two ways to go about this
a) like this pr does, return context on launch

or a different install process that not only receives the `ExEx` context but also a way to communicate this to the manager, imo this would be the more elegant approach.
for example, during install, you __create__ a subscription by providing the ExExHead as an argument, so that

https://github.com/paradigmxyz/reth/blob/516ae50320a1a8fecddd19b1685680fce325934c/crates/exex/exex/src/context.rs#L19-L33

is no longer created beforehand but the subscriptions will be created specific to the `ExExHead`

**Comment by shekhirin** [comment]
Going with the second approach in another PR

**Pull request closed without merging**